recipes/svector/all/conanfile.py

```diff
@@ -0,0 +1,64 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import get, copy
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.layout import basic_layout
+
+import os
+
+required_conan_version = ">=1.52.0"
+
+class PackageConan(ConanFile):
+    name = "svector"
+    description = "Compact SVO optimized vector for C++17 or higher"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/martinus/svector"
+    topics = ("vector", "container", "small-vector", "header-only")
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15.7",
+            "msvc": "191",
+            "gcc": "7",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.get_safe("compiler.version")) < minimum_version:
```

Comment:
```suggestion
        if self.info.settings.get_safe("compiler.cppstd"):
            check_min_cppstd(self, self._minimum_cpp_standard)
        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
        if minimum_version and Version(self.info.settings.get_safe("compiler.version")) < minimum_version:
```
As we are copying those headers, self.info.settings.compiler will be available when install the package, which is the most important

Reply:
@uilianries 
I have a question.
svector is header only library.
I think `self.info` is cleared by `self.info.clear()` in `package_id()`.
Is it wrong?

Reply:
@toge you are correct. Conan v2 separates our known `validate()` method by two: `validate()` and `build_validate()`. The new idea is validating when installing a package, but in case you need to validate only when building, you should use `build_validate` instead. For this specific case: `self.info.settings.compiler` is `None` because `self.info.clear()` is called, so `validate()` will do nothing when building the package, which is not important, as we are only coping header, not building anything for real. However, if you run `conan install` and try to install that pre-built package, `validate` will consider your profile, even if your package ID is for a header-only.

Reply:
@uilianries 
Thank you for precise explain!
Now I understand the reason use `self.info.settings` in `validate()`.
Thanks a lot!